files/src/main/scalajvm/sttp/tapir/files/TapirStaticContentEndpoints.scala

```diff
@@ -0,0 +1,281 @@
+          )
+        )
+      )
+      .out(
+        oneOf[StaticOutput[T]](
+          oneOfVariantClassMatcher(StatusCode.NotModified, emptyOutputAs(StaticOutput.NotModified), StaticOutput.NotModified.getClass),
+          oneOfVariantClassMatcher(
+            StatusCode.PartialContent,
+            body
+              .and(lastModifiedHeader)
+              .and(contentLengthHeader)
+              .and(contentTypeHeader)
+              .and(etagHeader)
+              .and(acceptRangesHeader)
+              .and(header[Option[String]](HeaderNames.ContentRange))
+              .map[StaticOutput.FoundPartial[T]](
+                (t: (T, Option[Instant], Option[Long], Option[MediaType], Option[ETag], Option[String], Option[String])) =>
+                  StaticOutput.FoundPartial(t._1, t._2, t._3, t._4, t._5, t._6, t._7)
+              )(fo => (fo.body, fo.lastModified, fo.contentLength, fo.contentType, fo.etag, fo.acceptRanges, fo.contentRange)),
+            classOf[StaticOutput.FoundPartial[T]]
+          ),
+          oneOfVariantClassMatcher(
+            StatusCode.Ok,
+            body
+              .and(lastModifiedHeader)
+              .and(contentLengthHeader)
+              .and(contentTypeHeader)
+              .and(etagHeader)
+              .and(contentEncodingHeader)
+              .map[StaticOutput.Found[T]]((t: (T, Option[Instant], Option[Long], Option[MediaType], Option[ETag], Option[String])) =>
+                StaticOutput.Found(t._1, t._2, t._3, t._4, t._5, t._6)
+              )(fo => (fo.body, fo.lastModified, fo.contentLength, fo.contentType, fo.etag, fo.contentEncoding)),
+            classOf[StaticOutput.Found[T]]
+          )
+        )
+      )
+  }
+
+  private lazy val staticHeadEndpoint: PublicEndpoint[HeadInput, StaticErrorOutput, HeadOutput, Any] = {
+    endpoint.head
+      .in(pathsWithoutDots.map[HeadInput](t => HeadInput(t))(_.path))
+      .errorOut(
+        oneOf[StaticErrorOutput](
+          oneOfVariantClassMatcher(
+            StatusCode.BadRequest,
+            emptyOutputAs(StaticErrorOutput.BadRequest),
+            StaticErrorOutput.BadRequest.getClass
+          ),
+          oneOfVariantClassMatcher(
+            StatusCode.NotFound,
+            emptyOutputAs(StaticErrorOutput.NotFound),
+            StaticErrorOutput.NotFound.getClass
+          )
+        )
+      )
+      .out(
+        oneOf[HeadOutput](
+          oneOfVariantClassMatcher(
+            StatusCode.Ok,
+            acceptRangesHeader
+              .and(contentLengthHeader)
+              .and(contentTypeHeader)
+              .map[HeadOutput.Found]((t: (Option[String], Option[Long], Option[MediaType])) => HeadOutput.Found(t._1, t._2, t._3))(fo =>
+                (fo.acceptRanges, fo.contentLength, fo.contentType)
+              ),
+            classOf[HeadOutput.Found]
+          )
+        )
+      )
+  }
+
+  implicit lazy val schemaForResourceRange: Schema[ResourceRange] = Schema(SchemaType.SBinary())
```

Comment:
this can go to Schema/Codec companion objects? probably will need to be a def for bincompat though